src/main/java/ch/hsr/geohash/BoundingBox.java

```diff
@@ -12,43 +12,115 @@
 
 public class BoundingBox implements Serializable {
 	private static final long serialVersionUID = -7145192134410261076L;
-	private double minLat;
-	private double maxLat;
-	private double minLon;
-	private double maxLon;
+	private double southLatitude;
+	private double northLatitude;
+	private double westLongitude;
+	private double eastLongitude;
+	private boolean intersects180Meridian;
 
 	/**
 	 * create a bounding box defined by two coordinates
 	 */
-	public BoundingBox(WGS84Point p1, WGS84Point p2) {
-		this(p1.getLatitude(), p2.getLatitude(), p1.getLongitude(), p2.getLongitude());
+	public BoundingBox(WGS84Point southWest, WGS84Point northEast) {
+		this(southWest.getLatitude(), northEast.getLatitude(), southWest.getLongitude(), northEast.getLongitude());
 	}
 
-	public BoundingBox(double y1, double y2, double x1, double x2) {
-		minLon = Math.min(x1, x2);
-		maxLon = Math.max(x1, x2);
-		minLat = Math.min(y1, y2);
-		maxLat = Math.max(y1, y2);
+	/**
+	 * Create a bounding box with the specified latitudes and longitudes. This constructor takes the order of the points into account.
+	 *
+	 * @param northLatitude
+	 * @param southLatitude
+	 * @param westLongitude
+	 * @param eastLongitude
+	 *
+	 * @throws IllegalArgumentException
+	 *             When the defined BoundingBox would go over one of the poles. This kind of box is not supported.
+	 */
+	public BoundingBox(double southLatitude, double northLatitude, double westLongitude, double eastLongitude) {
+		if (southLatitude > northLatitude)
+			throw new IllegalArgumentException("The southLatitude must not be greater than the northLatitude");
+
+		if (Math.abs(southLatitude) > 90 || Math.abs(northLatitude) > 90 || Math.abs(westLongitude) > 180 || Math.abs(eastLongitude) > 180) {
+			throw new IllegalArgumentException("The supplied coordinates are out of range.");
+		}
+
+		this.northLatitude = northLatitude;
+		this.westLongitude = westLongitude;
+
+		this.southLatitude = southLatitude;
+		this.eastLongitude = eastLongitude;
+
+		intersects180Meridian = eastLongitude < westLongitude;
 	}
 
+	/**
+	 * Clone constructor
+	 *
+	 * @param that
+	 */
 	public BoundingBox(BoundingBox that) {
-		this(that.minLat, that.maxLat, that.minLon, that.maxLon);
+		this(that.southLatitude, that.northLatitude, that.westLongitude, that.eastLongitude);
 	}
 
-	public WGS84Point getUpperLeft() {
-		return new WGS84Point(maxLat, minLon);
+	/**
+	 * Returns the NorthWestPoint of this BoundingBox as a new Point.
+	 *
+	 * @return
+	 */
+	public WGS84Point getNorthWestPoint() {
```

Comment:
Should we call this `northEastCorner()`?

Reply:
I presume `point` is also fine.

Reply:
I'd say its up to you. I liked point at the time of writing cause it clarifies what type will be returned. But that's also specified in the method header so yeah idc.

Corner is probably a little bit more clarifying if you call this method directly from the BoundingBox-object.

Shall we go with `northEastCornerPoint()`? That would be most clearly for me.

Do you want to emit the get in the method name or did you just forget to write it?